tests/test_bucket_manifest.py

```diff
@@ -0,0 +1,117 @@
+import pytest
+from unittest.mock import MagicMock
+from contextlib import contextmanager
+
+import boto3
+from botocore.stub import Stubber
+from botocore.exceptions import ClientError
+
+from batch_jobs.bucket_manifest.object_metadata_job import (
+    compute_object_metadata,
+    send_message,
+)
+from batch_jobs.bucket_manifest.bucket_manifest_job import (
+    get_messages_from_queue,
+    submit_job,
+    list_objects,
+)
+
+
+@contextmanager
+def mock_file(filepath, content=""):
+    with open(filepath, "w") as f:
+        f.write(content)
+    yield filepath
+    try:
+        os.remove(filepath)
+    except Exception:
+        pass
+
+
+def test_compute_object_metadata_success(monkeypatch, s3):
+
+    monkeypatch.setattr(
+        "batch_jobs.bucket_manifest.object_metadata_job.BUCKET", "test_bucket"
+    )
+    monkeypatch.setattr(
+        "batch_jobs.bucket_manifest.object_metadata_job.S3KEY", "test_key"
+    )
+
+    monkeypatch.setattr("batch_jobs.bucket_manifest.object_metadata_job.MAX_RETRIES", 1)
+    output = compute_object_metadata()
+    assert output == {
+        "url": "s3://test_bucket/test_key",
+        "md5": "d9673f3128fcfbd70d040f7dc18afbd8",
```

Comment:
is this from https://github.com/uc-cdis/aws-batch-jobs/pull/1/files#diff-484462fced51d1a06b1d93b4a44dd535R18? looks like size is 11 there, just curious where these fake values are coming from

Reply:
It is from `https://github.com/uc-cdis/aws-batch-jobs/blob/1c5a6aabd1e888edbfc23fc12e1955d1475d0908/tests/conftest.py#L49`

Yeah it is confusing. I will change the the value of md5 here `https://github.com/uc-cdis/aws-batch-jobs/blob/1c5a6aabd1e888edbfc23fc12e1955d1475d0908/tests/conftest.py#L18`

Reply:
wait, I'm still confused.  None the fake dicts in the conftest have a size of `7`, so how is this ouput asserting that successfully?

Reply:
this test uses the fixture `s3`. The fixture created a bucket `test_bucket` and upload a file with content `Awesome` and length(`Awesome`) == 7

Reply:
oh I totally understand your confusion. There are two separated tests. One is for test computing object metadata and one is for test consuming sqs. The former is setup with `s3` fixture that creates bucket and uploads an object to it, the latter is setup with `create_mock_sqs` fixture. To connect the two test, I think I should modify the `fake_message1` so that it represents the file uploaded by `s3` fixture

Reply:
ohhh, I see now